nalgebra-sparse/tests/unit_tests/coo.rs

```diff
@@ -87,6 +87,26 @@ fn coo_construction_for_valid_data() {
     }
 }
 
+#[test]
+fn coo_triplets_iter_mut() {
+    // Arbitrary matrix, with duplicates
+    let i = vec![0, 1, 0, 0, 0, 0, 2, 1];
+    let j = vec![0, 2, 0, 1, 0, 3, 3, 2];
+    let v = vec![2, 3, 4, 7, 1, 3, 1, 5];
+    let mut coo =
+        CooMatrix::<i32>::try_from_triplets(3, 5, i.clone(), j.clone(), v.clone()).unwrap();
+    coo.triplet_iter_mut().for_each(|(_i, _j, v)| *v = *v * *v);
```

Comment:
By ignoring the indices, we are not testing that the indices produced are correct (for example, they could be transposed!). The easiest way to do that is probably just to collect the mutable iterator, i.e. `triplet_iter_mut().collect()`, instead of `triplet_iter()` below and check the result against expected values.

Moreover, I think it'd be a good idea to manually write out the expected values rather than relying on programmatically producing them.